### ISSUE TITLE
Remove dependency on CurrentDirectory from all MS-owned tasks

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#3630]
We should clean up at least the tasks that we own to remove the dependency on CurrentDirectory.

Things like this:
https://referencesource.microsoft.com/#PresentationBuildTasks/BuildTasks/Microsoft/Build/Tasks/Windows/MarkupCompilePass1.cs,83

This would move use one step closer to being able to run targets in parallel where possible (currently not possible because tasks depend on Environment.CurrentDirectory and you can't have two different ones simultaneously).

Also it's good for reproducible builds, so that it doesn't matter where you're running the build from. Otherwise we can get stuff like this:

![image](https://user-images.githubusercontent.com/679326/44290641-80ecd480-a22e-11e8-9afa-7489de91c7cd.png)

**Comment by rainersigwald** [comment]
A thought occurred to me: we could make these easier to find by introducing an MSBuild flag `MSBUILDIntentionallyMangleCurrentDirectoryBeforeExecutingTasks` or similar, running builds with it, and filing bugs on the tasks.

**Comment by KirillOsenkov** [comment]
Yeah it's easy to find these things. Basically eyeball usages of Environment.CurrentDirectory and Directory.GetCurrentDirectory in our codebase to begin with.
